tests/unit/lib/charms/vault_k8s/v0/test_vault_tls.py

```diff
@@ -132,11 +132,11 @@ def test_given_not_leader_and_ca_not_set_when_configure_then_status_is_waiting(
             bind_address="1.2.1.2", ingress_address="10.1.0.1"
         )
 
-        self.harness.charm.on.config_changed.emit()
+        self.harness.evaluate_status()
```

Comment:
How about running both functions? I feel like we should check that configure returns properly too.

Applies to all the other changes below too.

Reply:
I only changed the tests where we were validating that the status was the right one. Since status is only maintained by collect status, I see no need to also call another event (ex. config_changed)